quipucords/api/scanjob/tests_scanjob.py

```diff
@@ -287,25 +287,127 @@ def test_details(self):
+                                'source_type': 'network',
+                                'most_recent_connect_scan': 1},
+                     'systems': [
+                         {'name': 'Foo',
+                          'status': 'success',
+                          'facts': [
+                              {'name': 'fact_key',
+                               'value': 'fact_value'}]}]}]}}
+        self.assertEqual(json_response, expected)
+
+    def test_connection(self):
+        """Get ScanJob connection results."""
+        # pylint: disable=no-member
+        scan_job, scan_task = create_scan_job(
+            self.source, ScanTask.SCAN_TYPE_INSPECT)
 
-        self.assertEqual(
-            json_response, {
-                'connection_results': {'task_results': [
-                    {'source':
-                     {'id': 1, 'name': 'source1', 'source_type': 'network',
-                      'most_recent_connect_scan': 1},
-                     'systems':
-                     [{'name': 'Foo', 'credential':
-                       {'id': 1, 'name': 'cred1'},
-                       'status': 'success'}]}]},
-                'inspection_results': {'task_results': [
-                    {'source':
-                     {'id': 1, 'name': 'source1', 'source_type': 'network',
-                      'most_recent_connect_scan': 1},
-                     'systems':
-                     [{'name': 'Foo', 'status': 'success',
-                       'facts': [
-                           {'name': 'fact_key', 'value': 'fact_value'}]}]}]}})
+        # Create a connection system result
+        sys_result = SystemConnectionResult(name='Foo',
+                                            source=self.source,
+                                            credential=self.cred,
+                                            status=SystemConnectionResult
+                                            .SUCCESS)
+        sys_result.save()
+        conn_result = scan_task.prerequisites.first().connection_result
+        conn_result.systems.add(sys_result)
+        conn_result.save()
+
+        url = reverse('scanjob-detail', args=(scan_job.id,)) + 'connection/'
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        json_response = response.json()
+        expected = {'count': 1,
+                    'next': None,
+                    'previous': None,
+                    'results': [{'name': 'Foo',
+                                 'status': 'success',
+                                 'credential': {'id': 1,
+                                                'name': 'cred1'},
+                                 'source': {'id': 1,
+                                            'name': 'source1',
+                                            'source_type': 'network'}}]}
+        self.assertEqual(json_response, expected)
+
+    def test_connection_failed_success(self):
```

Comment:
This test currently has 2 results with a single task.  

I think we should have a test with two scan tasks:
 - test total ordering cross tasks
- test case where one task has no results the other has results